### PR TITLE
Add command for basic parsing of HTML using BeautifulSoup

### PR DESCRIPTION
**Pull request by PlaceReporter99**

### CODE REVIEW

**Comment by PlaceReporter99** [comment]
Well, I need help resolving dependency issues…

**Review by Steffan153 — DISMISSED** [review]
Not the eval. That will be another security vulnerability.

File: vyxal/elements.py [hunk 1]
```diff
@@ -443,6 +444,10 @@ def process_element(
     "k□": process_element("[[0,1],[1,0],[0,-1],[-1,0]]", 0),
     "kṘ": process_element('"IVXLCDM"', 0),
     "k•": process_element('["qwertyuiop","asdfghjkl","zxcvbnm"]', 0),
+    "¨P": process_element(
+        'in = pop(stack, 2, ctx) ; stack.append(eval(f"bs({in[1]},"html.parser").{in[0]}"))',
```
Comment (Lyxal) [inline vyxal/elements.py:448]: This currently returns a `bs4.BeautifulSoup` object, which will can't be used by anything - the only types used in Vyxal are numbers (a combination of `int and any `Sympy` types), strings, lists/`LazyList`s and functions. The HTML parser would need to return something usable by programs.

**Review by Lyxal — CHANGES_REQUESTED** [review]

File: vyxal/elements.py (hunk 1 above)
Comment (PlaceReporter99) [reply vyxal/elements.py]: It will be converted to string before returning.

File: vyxal/elements.py [hunk 2]
```diff
@@ -445,7 +445,7 @@ def process_element(
     "kṘ": process_element('"IVXLCDM"', 0),
     "k•": process_element('["qwertyuiop","asdfghjkl","zxcvbnm"]', 0),
     "¨P": process_element(
-        'in = pop(stack, 2, ctx) ; stack.append(eval(f"bs({in[1]},"html.parser").{in[0]}"))',
+        'in = pop(stack, 2, ctx) ; stack.append(str(eval(f"bs({in[1]},"html.parser").{in[0]}")))',
```
Comment (Lyxal) [inline vyxal/elements.py:448]: Just converting to string returns the original HTML with slightly less newlines/somewhat neater formatting. It doesn't return any parsed information at all. You would need to return a list of all items in the HTML document.

**Review by Lyxal — CHANGES_REQUESTED** [review]

File: vyxal/elements.py [hunk 3]
```diff
@@ -443,6 +444,10 @@ def process_element(
     "k□": process_element("[[0,1],[1,0],[0,-1],[-1,0]]", 0),
     "kṘ": process_element('"IVXLCDM"', 0),
     "k•": process_element('["qwertyuiop","asdfghjkl","zxcvbnm"]', 0),
+    "¨P": process_element(
```
[the remark this answers is not in the document]
Comment (PlaceReporter99) [reply vyxal/elements.py]: Fixed.

File: vyxal/elements.py (hunk 2 above)
Comment (PlaceReporter99) [reply vyxal/elements.py]: Now returns list with actual parse data.

**Comment by PlaceReporter99** [comment]
> Not the eval. That will be another security vulnerability.

fixed

**Comment by PlaceReporter99** [comment]
Is this ready to merge?

**Comment by Lyxal** [comment]
> Is this ready to merge?

@PlaceReporter99 I was wondering if you found the changes I made to the PR were what you had in mind.

**Comment by PlaceReporter99** [comment]
Well I was thinking where you specify the HTML document and the parse syntax and it returns that section of the html document, like this:
<code><code>\`HTML Doc\`\`body.div.form.p\`¨P</code> (Vyxal) --> <code>stack.append(flattenlist(list(BeautifulSoup("HTML Doc","html.parser").find_all("body").find_all("div").find_all("form").find_all("p"))))</code> (Pseudo-Python)</code>
Basically extracts all paragraphs that are in a form that are in a div that are in a body.

**Review by Lyxal — APPROVED** [review]

**Pull request merged**